test/hotspot/jtreg/runtime/os/TestTracePageSizes.java

```diff
@@ -93,32 +97,87 @@
 public class TestTracePageSizes {
     // Store address ranges with known page size.
     private static LinkedList<RangeWithPageSize> ranges = new LinkedList<>();
-    private static boolean debug;
+    private static boolean debug = false;
+    private static int run = 0;
+
+    // Copy smaps locally
+    // (To minimize chances of concurrent modification when parsing, as well as helping with error analysis)
+    private static Path copySmaps() throws Exception {
+        Path p1 = Paths.get("/proc/self/smaps");
+        Path p2 = Paths.get("smaps-copy-" +  ProcessHandle.current().pid() + "-" + run++ + ".txt");
+        Files.copy(p1, p2, StandardCopyOption.REPLACE_EXISTING);
+        System.out.println("Copied " + p1.getFileName() + " to " + p2.getFileName() + "...");
```

Comment:
`Path.toString()` is defined, I think, so...

```suggestion
        System.out.println("Copied " + p1 + " to " + p2 + "...");
```